clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2129,63 +2167,183 @@ private Map<String, KafkaFuture<TopicDescription>> handleDescribeTopicsByNames(f
             }
         }
         final long now = time.milliseconds();
-        Call call = new Call("describeTopics", calcDeadlineMs(now, options.timeoutMs()),
-            new LeastLoadedNodeProvider()) {
 
-            private boolean supportsDisablingTopicCreation = true;
+        if (options.useDescribeTopicsApi()) {
+            RecurringCall call = new RecurringCall("DescribeTopics-Recurring", calcDeadlineMs(now, options.timeoutMs()), runnable) {
```

Comment:
Instead of building the recurring call framework, can we just schedule the next call recursively?  Something like 

```
  Call call = new Call("describeTopics", calcDeadlineMs(now, options.timeoutMs()), new LeastLoadedNodeProvider()) {
    DescribeTopicPartitionsRequestData.Cursor currentCursor = new DescribeTopicPartitionsRequestData.Cursor();
    // <...>
    @Override
    void handleResponse(AbstractResponse abstractResponse) {
      // ... Do the needful ...
      // ...
       if (hasMore) {
         // ... Set new cursor ...
         // ...
         runnable.call(this, now);
       }
    }
    // <...>
  }

Reply:
Thanks for the advice. We still want to avoid loading a huge topic in the memory but the current implementation does not really achieve that. I will make a change to address this.
Maybe it is a bit over-engineering to have more than 1 ongoing call in the pipeline. I will keep at most one call in the next iteration commit.

Reply:
I think we can defer pipelining for now. If we do it the "dumb" way of just loading the next page of results as we drain the iterator, it will be slower, but it will work.

Reply:
Not doing pipelining in the first version sounds good to me.  We can optimize later if needed.